src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasServer.kt

```diff
@@ -39,23 +40,29 @@ class CosmasServer(port: Int) {
     }
 
     private fun stop() {
-        server.shutdown()
+        this.server.shutdown()
     }
 
     fun blockUntilShutDown() {
-        server.awaitTermination()
+        this.server.awaitTermination()
     }
 }
 
 fun main(args: Array<String>) {
     val arg = CosmasServerArgs(ArgParser(args))
     println("Try to bind in port ${arg.port}")
-    val server = CosmasServer(arg.port)
+    val server =
+            if (arg.bucket != "") {
+                CosmasServer(arg.port, CosmasGoogleCloudService(arg.bucket))
+            } else {
+                CosmasServer(arg.port, CosmasInMemoryService())
+            }
     println("Start working in port ${arg.port}")
     server.start()
     server.blockUntilShutDown()
 }
 
 class CosmasServerArgs(parser: ArgParser) {
-    val port: Int by parser.storing("--port", help = "choose port") { toInt() }.default { 50051 }
+    val port: Int by parser.storing("--port", help = "choose port that server will hear") { toInt() }.default { 50051 }

```

Comment:
programmers usually use "listen to" when talking about ports

Reply:
done